src/Plugins/ComplexCore/src/ComplexCore/Filters/GenerateColorTableFilter.cpp

```diff
@@ -0,0 +1,98 @@
+#include "GenerateColorTableFilter.hpp"
+
+#include "Algorithms/GenerateColorTable.hpp"
+
+#include "complex/DataStructure/DataPath.hpp"
+#include "complex/DataStructure/IDataArray.hpp"
+#include "complex/Filter/Actions/CreateArrayAction.hpp"
+#include "complex/Parameters/ArrayCreationParameter.hpp"
+#include "complex/Parameters/ArraySelectionParameter.hpp"
+#include "complex/Parameters/GenerateColorTableParameter.hpp"
+
+using namespace complex;
+
+namespace complex
+{
+//------------------------------------------------------------------------------
+std::string GenerateColorTableFilter::name() const
+{
+  return FilterTraits<GenerateColorTableFilter>::name.str();
+}
+
+//------------------------------------------------------------------------------
+std::string GenerateColorTableFilter::className() const
+{
+  return FilterTraits<GenerateColorTableFilter>::className;
+}
+
+//------------------------------------------------------------------------------
+Uuid GenerateColorTableFilter::uuid() const
+{
+  return FilterTraits<GenerateColorTableFilter>::uuid;
+}
+
+//------------------------------------------------------------------------------
+std::string GenerateColorTableFilter::humanName() const
+{
+  return "Generate Color Table";
+}
+
+//------------------------------------------------------------------------------
+std::vector<std::string> GenerateColorTableFilter::defaultTags() const
+{
+  return {"#Core", "#Image"};
+}
+
+//------------------------------------------------------------------------------
+Parameters GenerateColorTableFilter::parameters() const
+{
+  Parameters params;
+
+  // Create the parameter descriptors that are needed for this filter
+  params.insertSeparator({"Input Parameters"});
+  params.insert(std::make_unique<GenerateColorTableParameter>(k_SelectedPreset_Key, "Select Preset...", "", ""));
+  params.insertSeparator({"Required Data Objects"});
+  params.insert(std::make_unique<ArraySelectionParameter>(k_SelectedDataArrayPath_Key, "Data Array", "", DataPath{}, complex::GetAllDataTypes(), ArraySelectionParameter::AllowedComponentShapes{{1}}));
+  params.insertSeparator({"Created Data Objects"});
+  params.insert(std::make_unique<ArrayCreationParameter>(k_RgbArrayPath_Key, "Output RGB Array", "", DataPath{}));
```

Comment:
```suggestion
  params.insert(std::make_unique<ArrayCreationParameter>(k_RgbArrayPath_Key, "Output RGB Array", "The rgb array created by normalizing each element of the input array and converting to a color based on the selected preset color scheme", DataPath{}));
```

Reply:
@joeykleingers  Feel free to tweek any of these help texts if I got something wrong